src/java.base/share/classes/java/nio/file/FileTreeWalker.java

```diff
@@ -198,14 +198,12 @@ IOException ioeException() {
      * the walk is following sym links is not. The {@code canUseCached}
      * argument determines whether this method can use cached attributes.
      */
-    @SuppressWarnings("removal")
     private BasicFileAttributes getAttributes(Path file, boolean canUseCached)
         throws IOException
     {
         // if attributes are cached then use them if possible
         if (canUseCached &&
-            (file instanceof BasicFileAttributesHolder) &&
-            (System.getSecurityManager() == null))
+            (file instanceof BasicFileAttributesHolder))
```

Comment:
We can replace this with `if ((canUseCached && (file instanceof BasicFileAttributesHolde holder) {` and it would make it a lot more readable.